src/main/resources/com/google/api/codegen/java/settings.snip

```diff
@@ -79,6 +79,12 @@
 
   private static final String DEFAULT_GAPIC_NAME = "gapic";
   private static final String DEFAULT_GAPIC_VERSION = "";
+  private static final String ARTIFACT_ID = "{@xsettingsClass.artifactId}";
+  private static final String GROUP_ID = "com.google.cloud";
+  private static final String META_FILE_ROOT = "/META-INF/maven/";
```

Comment:
Our generated code will most likely be used in google-cloud-java package which uses maven. I can make it default to use the `getImplementationVersion()` approach and fall back to use the maven version.